packages/db/tests/lib/test-dupliate-detection.js

```diff
@@ -0,0 +1,221 @@
+  granulePgModel,
+  count,
+  collectionCumulusId,
+  producerGranuleId,
+  granuleIdPrefix = 'g',
+}) => await granulePgModel.insert(
+  knex,
+  Array.from({ length: count }, (_, i) =>
+    fakeGranuleRecordFactory({
+      granule_id: `${granuleIdPrefix}-${i}`,
+      producer_granule_id: producerGranuleId,
+      collection_cumulus_id: collectionCumulusId,
+    }))
+);
+
+// Mark the provided granules as active or inactive in the granule_groups table.
+const setGranuleGroupStates = async (knex, granules, state) => {
+  const granuleGroupsPgModel = new GranuleGroupsPgModel();
+  await Promise.all(
+    granules.map((g) =>
+      granuleGroupsPgModel.create(knex, {
+        granule_cumulus_id: g.cumulus_id,
+        state,
+      }))
+  );
+};
+
+test.before(async (t) => {
+  t.context.testDbName = `granule_${cryptoRandomString({ length: 10 })}`;
+  const { knexAdmin, knex } = await generateLocalTestDb(t.context.testDbName, migrationDir);
+  t.context.knexAdmin = knexAdmin;
+  t.context.knex = knex;
+  t.context.granulePgModel = new GranulePgModel();
+  t.context.collectionPgModel = new CollectionPgModel();
+
+  const collections = [
+    { name: 'fakeCollection', version: 'v1' },
+    { name: 'testCollection2', version: 'v2' },
+  ];
+
+  for (const [i, { name, version }] of collections.entries()) {
+    const collectionId = constructCollectionId(name, version);
+    const testPgCollection = fakeCollectionRecordFactory({ name, version });
+
+    /* eslint-disable-next-line no-await-in-loop */
+    const [pgCollection] = await t.context.collectionPgModel.create(knex, testPgCollection);
+
+    // Save to context, when i is 0, nothing is appended
+    t.context[`collectionId${i || ''}`] = collectionId;
+    t.context[`testPgCollection${i || ''}`] = testPgCollection;
+    t.context[`collectionCumulusId${i || ''}`] = pgCollection.cumulus_id;
+  }
+});
+
+test.after.always(async (t) => {
+  await destroyLocalTestDb(t.context);
+});
+
+test('getNextGranuleGroupId returns increasing sequence values', async (t) => {
+  const knex = t.context.knex;
+
+  const id1 = await getNextGranuleGroupId(knex);
+  const id2 = await getNextGranuleGroupId(knex);
+
+  t.true(Number.isInteger(id1));
+  t.true(Number.isInteger(id2));
+  t.true(id2 > id1, 'Second call should return a higher group_id');
+});
+
+test('findActiveDuplicateGranules finds multiple duplicates in same collection', async (t) => {
+  const { knex, collectionId, collectionCumulusId, granulePgModel } = t.context;
+
+  const producerGranuleId = 'same-producer-id-same-collection';
+
+  await insertGranulesWithProducerId({
+    knex,
+    granulePgModel,
+    count: 3,
+    collectionCumulusId,
+    producerGranuleId,
+    granuleIdPrefix: 'same-coll',
+  });
+
+  const result = await findActiveDuplicateGranules({
+    collectionId,
+    producerGranuleId,
+    collectionCumulusId,
+  }, knex);
+
+  t.is(result.sameCollectionMatches.length, 3);
+  t.is(result.differentCollectionMatches.length, 0);
```

Comment:
Nit: 

We're not explicitly calling out the other default behaviors here (things like customCriteriaMatches always being 0).   Given its' a feature that seems mostly ok, but there's a possible concern that when an update is made we neglect to address all existing test cases.       I'm ok leaving this for our review process ,but could also see it falling to being explicit in this PR as well.